tests/tools/onert_train/src/onert_train.cc

```diff
@@ -59,19 +59,96 @@ int main(const int argc, char **argv)
 
     // prepare measure tool
     Measure measure(args.getMemoryPoll());
+    nnfw_train_info tri;
 
     nnfw_session *session = nullptr;
     NNPR_ENSURE_STATUS(nnfw_create_session(&session));
 
     // ModelLoad
+    bool tinfo_loaded = false;
+    auto model_file = args.getModelFilename();
+    auto model_extension = model_file.substr((model_file.find_last_of(".") + 1));
+
     measure.run(PhaseType::MODEL_LOAD, [&]() {
-      if (args.useSingleModel())
-        NNPR_ENSURE_STATUS(
-          nnfw_load_model_from_modelfile(session, args.getModelFilename().c_str()));
+      if (args.useSingleModel() && model_extension == "circle+")
+        nnfw_load_training_model_from_modelfile(session, model_file.c_str(), &tri);
+      else if (args.useSingleModel())
+        NNPR_ENSURE_STATUS(nnfw_load_model_from_file(session, model_file.c_str()));
       else
-        NNPR_ENSURE_STATUS(nnfw_load_model_from_file(session, args.getPackageFilename().c_str()));
+        NNPR_ENSURE_STATUS(nnfw_load_model_from_file(session, model_file.c_str()));
     });
 
+    // If failed to load train_info from model file,
+    // or train_info is given through arugment
+    if (tinfo_loaded == false || args.isTrainingInfo())
```

Comment:
This `tinfo_loaded` is always `false` value. Is it your intention?

Reply:
Ah, that was my mistake.
updated!